From 38e133638720a784d929143b4eff8b4b8765884e Mon Sep 17 00:00:00 2001
From: Christian Brauner <christian.brauner@ubuntu.com>
Date: Mon, 11 Sep 2017 03:30:00 +0200
Subject: storage: avoid segfault

When the "lxc.rootfs.path" property is not set and users request a container
copy we would segfault since strstr() would be called on a NULL pointer.

Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>
---
 src/lxc/storage/storage.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/lxc/storage/storage.c b/src/lxc/storage/storage.c
index c8909123..708d070a 100644
--- a/src/lxc/storage/storage.c
+++ b/src/lxc/storage/storage.c
@@ -337,6 +337,11 @@ struct lxc_storage *storage_copy(struct lxc_container *c, const char *cname,
 	struct rsync_data data = {0};
 	char cmd_output[MAXPATHLEN] = {0};
 
+	if (!src) {
+		ERROR("No rootfs specified");
+		return NULL;
+	}
+
 	/* If the container name doesn't show up in the rootfs path, then we
 	 * don't know how to come up with a new name.
 	 */
